### PR TITLE
Fast reject expensive optimizations for passes with zero number of locals

### PR DESCRIPTION
**Pull request by MaxGraey**
It seems some optimization passes which manipulate only locals could be early rejected when number of locals is zero

### CODE REVIEW

File: src/passes/ReorderLocals.cpp [hunk 1]
```diff
@@ -54,10 +58,9 @@ struct ReorderLocals : public WalkerPass<PostWalker<ReorderLocals>> {
     // Gather information about local usages.
     walk(curr->body);
     // Use the information about local usages.
-    std::vector<Index> newToOld;
-    for (size_t i = 0; i < num; i++) {
-      newToOld.push_back(i);
-    }
+    std::vector<Index> newToOld((size_t)num);
+    // Fill newToOld from "0" to "num".
+    std::iota(newToOld.begin(), newToOld.end(), 0);
```
Comment (tlively) [inline src/passes/ReorderLocals.cpp:63]: This seems like an unrelated change. What is the advantage of using `std::iota` here?

File: src/passes/ReorderLocals.cpp [hunk 2]
```diff
@@ -131,9 +132,10 @@ struct ReorderLocals : public WalkerPass<PostWalker<ReorderLocals>> {
     auto oldLocalIndices = curr->localIndices;
     curr->localNames.clear();
     curr->localIndices.clear();
+    auto end = oldLocalNames.end();
```
Comment (tlively) [inline src/passes/ReorderLocals.cpp:135]: I'm not sure introducing a new variable is worth doing here.

File: src/passes/ReorderLocals.cpp (hunk 1 above)
Comment (MaxGraey) [reply src/passes/ReorderLocals.cpp:63]: Using one line method similar to std::fill? But perhaps `std::iota` is quite exotic for many peoples. I could revert this

**Comment by MaxGraey** [comment]
> Does this change have any noticeable impact on runtimes? I expect that functions with with no locals typically take negligible time in these passes anyways.

I don't measure perf impact, but if we take ReorderLocals pass for example which handle mix of locals and params, sort this and only after that segregate locals/params via `curr->isParam` with coping param's indexes as is. So when we have functions with zero locals but some input params it could skip all this process
